tools/tcpconnect.py

```diff
@@ -322,6 +322,31 @@
     return 0;
 }
 
+#include <uapi/linux/udp.h>
+
+int trace_udpv6_recvmsg(struct pt_regs *ctx)
+{
+    struct sk_buff *skb = (struct sk_buff *)PT_REGS_PARM2(ctx);
+    struct udphdr *hdr = (void*)skb->head + skb->transport_header;
+    struct dns_data_t *event;
+    int zero = 0;
+    void *data;
+
+    /* hex(53) = 0x0035, htons(0x0035) = 0x3500 */
+    if (hdr->source != 0x3500)
+        return 0;
+
+    /* skip UDP header */
+    data = skb->data + 8;
+
+    event = dns_data.lookup(&zero);
+    if (!event)
+        return 0;
+
+    bpf_probe_read(event->pkt, sizeof(event->pkt), data);
```

Comment:
Could we have some garbage here? Is it possible we check the udp packet size and only copy min(udp_size, sizeof(event->pkt))?

Reply:
Do you mean something like this ?

1st record(8 bytes): XXXX XXXX
2nd record(6 bytes): YYYY YYXX    <- the last two bytes is from last record

It could be.

But DNS resource record format has a field specifying the record length and dnslib parses this correctly (unless the record is truncated).

Reply:
Okay. that is fine then since the data is parsed correctly in user space library.